packages/components/src/components/Calculator/CalculatorResult.tsx

```diff
@@ -38,6 +51,15 @@ export const CalculatorResult: FC<Props> = ({
     SqValueResult | undefined
   >(() => savedState?.calculatorResult);
 
+  const valueResultViewer = useMemo(() => {
+    return (
+      !!calculatorResult &&
+      _settings && (
+        <ValueResultViewer result={calculatorResult} settings={_settings} />
+      )
+    );
+  }, [calculatorResult, _settings]);
```

Comment:
The problem was that React was doing a poor comparison of settings, and frequently updating. This code does a deep compare and ``useMemo`` to make sure it doesn't update unless actually needed, by settings or the result

Reply:
I imagine that this solution would still have the deep comparison cost, just bring it up higher for it to be more useful. Seems better than it is now (am changing), but I'm still curious if it's really the best thing in React.

Reply:
I tried this in the new commit, let me know if that works better.

Reply:
> I imagine that this solution would still have the deep comparison cost

Not really. `useMemo` never does deep comparison or even shallow object comparison, only identity checks.

Another thing: `useMemo` for constructing new elements, like you did here, is _probably_ ok, but I'm not 100% sure and I've rarely seen people do that.

Instead, you could wrap `<ValueResultViewer>` in [memo](https://react.dev/reference/react/memo). Then `<ValueResultViewer>` won't re-render when its props don't change, and `calculatorResultSettings` won't change if `settings` didn't change.

Reply:
One thing I like about using useMemo() for the elements, is that it's possible they might be used for other components, so this could be a bit more generic.

Reply:
I'm not sure what you mean; wrapping `useMemo` in a custom hook?

`useMemo` is more generic in a sense that it could be used for memoizing not just JSX but anything else, but `memo` is the official and more idiomatic way of memoizing components specifically, with more readable syntax and with an approach that allows you to make a decision on a component level.

See also:
- https://stackoverflow.com/a/62337860
- https://stackoverflow.com/a/67096926

(I can't find an official opinion of this sort, but this is what I've usually seen: returning JSX from hooks is possible but discouraged)

Reply:
Ah, sorry, I misunderstood you above. 

Do you think I should change this code further now, or should we merge?

Reply:
It's not that big of a deal but it's better if you change it now, shouldn't be hard.

One thing to pay attention to, with `memo`, is that you should do `const MyComponent = memo(function MyComponent(...: Props) { ... })`, not `const MyComponent = memo((...: Props) => { ... })`, because the second version doesn't assign a name to the function and it shows in devtools incorrectly.